### PR TITLE
Identify tool for 3D view

### PR DESCRIPTION
**Pull request by wonder-sk**
Adds a map tool for identification of features within 3D map view. It reuses existing 2D map tool for identification, that means the results dock widget is shared among the 2D and 3D map.

Current limitations:
- ignores 3D entities created by map layers (i.e. only identifies features on the map on the terrain)
- does not show highlight in the 3D view
- assumes 3D view has the same CRS and search radius as 2D map canvas

These limitations will be removed in followup pull requests.

Financed by Department of Environment, Land and Infrastructure Engineering (DIATI) and Department of Architecture and Design (DAD) of Politecnico di Torino within ResCult project  (fund: European Commission, DG-ECHO Humanitarian Aid And Civil Protection)  in collaboration with Faunalia.

### CODE REVIEW

File: src/app/3d/qgs3dmapcanvas.cpp [hunk 1]
```diff
@@ -108,3 +109,53 @@ void Qgs3DMapCanvas::saveAsImage( const QString fileName, const QString fileForm
...
+  {
+    mEngine->window()->removeEventFilter( this );
+    mScene->cameraController()->setEnabled( true );
+    mEngine->window()->setCursor( Qt::OpenHandCursor );
+  }
+  else if ( !mMapTool && tool )
+  {
+    mEngine->window()->installEventFilter( this );
+    mScene->cameraController()->setEnabled( false );
+    mEngine->window()->setCursor( Qt::CrossCursor );
```
Comment (m-kuhn) [inline src/app/3d/qgs3dmapcanvas.cpp]: Should this call a (virtual) method on the tool to determine the cursor?

**Comment by m-kuhn** [comment]
Nice!

File: src/app/3d/qgs3dmapcanvas.cpp (hunk 1 above)
Comment (wonder-sk) [reply src/app/3d/qgs3dmapcanvas.cpp]: At some point yes when we need different cursors for different tools...

**Comment by ghtmtt** [comment]
thanks @wonder-sk! Amazing feature

File: src/app/3d/qgs3dmapcanvas.cpp (hunk 1 above)
Comment (m-kuhn) [reply src/app/3d/qgs3dmapcanvas.cpp]: It just looks like code that belongs into a specific map tool (which already exists) and not into the canvas...

Comment (wonder-sk) [reply src/app/3d/qgs3dmapcanvas.cpp]: cf87b85

**Comment by wonder-sk** [comment]
Oops only now I have realized I have pushed my branch to qgis repo instead of my repo - will remove the branch once merged.

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**